pkg/block/indexheader/binary_reader.go

```diff
@@ -520,84 +520,79 @@ func newFileBinaryReader(path string, postingOffsetsInMemSampling int) (bw *Bina
 		return nil, errors.Wrap(err, "read symbols")
 	}
 
-	var lastKey []string
+	var lastName, lastValue []byte
 	if r.indexVersion == index.FormatV1 {
 		// Earlier V1 formats don't have a sorted postings offset table, so
 		// load the whole offset table into memory.
 		r.postingsV1 = map[string]map[string]index.Range{}
 
 		var prevRng index.Range
-		if err := index.ReadOffsetTable(r.b, r.toc.PostingsOffsetTable, func(key []string, off uint64, _ int) error {
-			if len(key) != 2 {
-				return errors.Errorf("unexpected key length for posting table %d", len(key))
+		if err := index.ReadPostingsOffsetTable(r.b, r.toc.PostingsOffsetTable, func(name, value []byte, postingsOffset uint64, _ int) error {
```

Comment:
Is this taken from Prometheus?

Reply:
It's "inspired" by this: https://github.com/prometheus/prometheus/blob/3b4cbf8da4c5b9b8c6c2a0a30ae182dfbb0b8213/tsdb/index/index.go#L1166-L1217